src/Core/Silk.NET.Core/Loader/LibraryLoader.cs

```diff
@@ -353,6 +353,11 @@ private class NetNextNativeLibraryLoader : LibraryLoader
         {
             protected override nint CoreLoadNativeLibrary(string name)
             {
+#if NET7_0_OR_GREATER
+                if (name == null)
```

Comment:
I'm not convinced this will work, or at least not work in a way that results in approval from Apple in the App Store. This is because all Vulkan calls will still be going through the dynamic loader (albeit to load functions from itself). I think you'll have to add what is called a `PInvokeOverride` which will force our loader to use `DllImport` and thereby directly link the appropriate functions. You can see an example of this here: https://github.com/dotnet/Silk.NET/blob/main/src/Windowing/Silk.NET.SDL/Sdl.cs#L14

Reply:
Yes, I have not yet been able to check whether Apple will allow this on their review.
I first tried using dylib but The Khronos Group does not provide them for the Simulator and Mac Catalyst. Using the PInvokeOverride attribute is the second thing I tried. But with it I can't compile for Mac and Android (an error at the linking stage). I think it is necessary to surround these generated classes with Defines. But in other cases, this option works.

Reply:
I've just checked the runtime source code and GetMainProgramHandle uses `dlopen`. libdl is banned by Apple on iOS, and apps using these symbols will not pass their review.

> Using the PInvokeOverride attribute is the second thing I tried. But with it I can't compile for Mac and Android (an error at the linking stage).

That's odd that Android is affected, perhaps we need to use `#if __IOS__` on the attribute and add a `net7.0-ios` TFM on Vulkan... Might not be the worst idea anyway, and have us automatically ProjectReference the new native package so users automatically download this.

But I know what's causing this, can you send me the compilation errors? Chances are we need to make some functions exempt from this override so we don't try and link them, as we had to do this with SDL [here](https://github.com/dotnet/Silk.NET/blob/main/generator.json#L881)

After those changes have been made BuildTools need to be rerun by `dotnet run --project src/Core/Silk.NET.BuildTools/Silk.NET.BuildTools.csproj -- generator.json --profile=Vulkan`. This will change the generated code.

If you need any help with any of this or would rather not do this, please let me know - this is quite nitty-gritty Silk.NET stuff which I'd imagine you didn't think you were signing up for! I'm also always happy to talk anything over here or in the Silk.NET Discord.

Reply:
Rework it to PInvokeOverride. But I encountered a problem with incorrect generation by SilkTouch. So I add new attribute PreventSilkTouchBugAttribute).

I also made sure that this is not a Roslyn problem: https://github.com/dotnet/roslyn/issues/59496
Perhaps it can be solved in SilkTouch simply by not using the all old class modifiers, but by taking only partial.